cellpack/autopack/loaders/recipe_loader.py

```diff
@@ -119,10 +107,49 @@ def _request_sub_recipe(self, inode):
         return data
 
     @staticmethod
-    def _migrate_version(recipe):
-        if "format_version" not in recipe:
-            recipe["bounding_box"] = recipe["options"]["boundingBox"]
-        return recipe
+    def _convert_to_representations(old_ingredient):
+        representations = RecipeLoader.default_values["representations"].copy()
+        if "sphereFile" in old_ingredient and old_ingredient["sphereFile"] is not None:
+            representations["packing"] = create_file_info_object_from_full_path(
+                old_ingredient["sphereFile"]
+            )
+        if "meshFile" in old_ingredient and old_ingredient["meshFile"] is not None:
+            representations["mesh"] = create_file_info_object_from_full_path(
+                old_ingredient["meshFile"]
+            )
+            if (
+                "coordsystem" in old_ingredient
+                and old_ingredient["coordsystem"] is not None
+            ):
+                representations["mesh"]["coordinate_system"] = old_ingredient[
+                    "coordsystem"
+                ]
+        if "pdb" in old_ingredient and old_ingredient["pdb"] is not None:
+            if ".pdb" in old_ingredient["pdb"]:
+                representations["atomic"] = {
+                    "path": "default",
+                    "name": old_ingredient["pdb"],
+                    "format": ".pdb",
+                }
+            else:
+                representations["atomic"] = {
+                    "id": old_ingredient["pdb"],
+                    "format": ".pdb",
+                }
+            if "source" in old_ingredient and "transform" in old_ingredient["source"]:
+                representations["atomic"]["transform"] = old_ingredient["source"][
+                    "transform"
+                ]
+
+        return representations
+
+    @staticmethod
+    def _convert_to_partners(old_ingredient):
+        partners = {}
+        for attribute in old_ingredient:
```

Comment:
Even though this isn't a huge amount of data it's generally better to not loop through data more that once if you don't need to. In line 157 we are already looping over all the attributes